docs/03-04-function-configuration-file.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Function's configuration file
+type: Details
+---
+
+When you initialize a Function (`init`), CLI creates the `config.yaml` file in your workspace folder. This file contains the whole Function's configuration and specification not only for the Function custom resource but also any other related resources you create for it, such as Subscriptions and API Rules.
+
+## Specification
+
+See the sample `config.yaml`:
+
+```yaml
+name: function-practical-filip5
+namespace: testme
+runtime: nodejs12
+source:
+    sourceType: inline
+    sourcePath: /tmp/cli
+subscriptions:
+    - name: function-practical-filip5
+      protocol: ""
+      filter:
+        dialect: ""
+        filters:
+            - eventSource:
+                property: source
+                type: exact
+                value: ""
+              eventType:
+                property: type
+                type: exact
+                value: sap.kyma.custom.demo-app.order.created.v1
+apiRules:
+    - name: function-practical-filip5
+      service:
+        host: path.34.90.136.181.xip.io
+      rules:
+        - methods:
+            - GET
+            - POST
+            - PUT
+            - PATCH
+            - DELETE
+            - HEAD
+          accessStrategies: []
+        - path: /path1/something1
+          methods:
+            - PUT
+            - PATCH
+            - DELETE
+          accessStrategies:
+            - handler: noop
+        - path: /path1/something2
+          methods:
+            - GET
+            - HEAD
+          accessStrategies:
+            - config:
+                requiredScope:
+                    - asd1.asd2.asd3
+                    - scope1
+                    - scope2
+              handler: oauth2_introspection
+        - path: /path2
+          methods:
+            - DELETE
+          accessStrategies:
+              handler: jwt
+            - config:
+                jwksUrls:
+                    - http://dex-service.kyma-system.svc.cluster.local:5556/keys
+                    - http://dex-service.kyma-system.svc.cluster.local:5556
+                trustedIssuers:
+                    - https://dex.34.90.136.181.xip.io
+                    - https://dex.34.90.136.181.xip.io
+env:
+    - name: REDIS_PASS
+      value: YgJUg8z6eA
+    - name: REDIS_PORT
+      value: "6379"
+    - name: REDIS_HOST
+      value: hb-redis-enterp-6541066a-edbc-422f-8bef-fafca0befea8-redis.testme.svc.cluster.local
+```
+
+## Parameters
+
+See all parameter descriptions:
+
+| Parameter         | Required | Related custom resource | Description                                   |
+| ---------------------------------------- | :------------: | ---------| ---------|
```

Comment:
Some values in the config.json, during the applying process, are defaulted by the Cli. Maybe should we add another section or column that can help users to understand what will be applied to the cluster if they missed some non-required fields? For example when the value of the `apiRules.service.port` field is empty then Cli will create apiRule with `80` as the value of the service port